l5kit/l5kit/rasterization/semantic_rasterizer.py

```diff
@@ -1,73 +1,55 @@
+from collections import defaultdict
 from typing import List, Optional, Tuple
 
 import cv2
 import numpy as np
-import pymap3d as pm
 
-from ..geometry import rotation33_as_yaw, transform_point
+from ..data.filter import filter_tl_faces_by_status
+from ..data.map_api import MapAPI
+from ..geometry import rotation33_as_yaw, transform_point, transform_points, world_to_image_pixels_matrix
 from .rasterizer import Rasterizer
 
+# sub-pixel drawing precision constants
+CV2_SHIFT = 8  # how many bits to shift in drawing
+CV2_SHIFT_VALUE = 2 ** CV2_SHIFT
 
-def render_semantic_map(
-    semantic_map: dict, x: float, y: float, yaw: float, raster_size: Tuple[int, int], pixel_size: np.ndarray
-) -> np.ndarray:
-    """Renders the semantic map at given x,y coordinate with rotation ``yaw``.
+
+def elements_within_bounds(center: np.ndarray, bounds: np.ndarray, half_extent: float) -> np.ndarray:
```

Comment:
What coordinates does this use? Can you please describe? Should this be in MapAPI perhaps? Otherwise, some geometry module would be more fitting for this method.

Reply:
geometry may be the right place actually. Coordinates system don't really matter here, it basically computes whether two rectangles intersect